frontend/src/components/board/CreatePostModal.jsx

```diff
@@ -0,0 +1,106 @@
+import { useState, useEffect } from "react";
+import "./CreatePostModal.css";
+
+const DEFAULT_GIF = "Enter GIF URL";
+
+//   author      String
+//   imageURL    String
+//   upvotes     Int
+//   boardId     Int
+//   title       String
+//   description String
+
+const CreatePostModal = ({ setCreatePostVisibility }) => {
+  const [formData, setFormData] = useState({
+    title: "",
+    description: "",
+    author: "",
+    imageURL: "",
+    boardId: "",
+  });
+
+  const doSomethingElse = () => {
+    console.log("somthing");
+  };
+
+  const handleChange = (event) => {
```

Comment:
You might want some quick validation on GIF URLs here! Or maybe even a separate function for handling GIF changes. It's hard to prevent bad URLs entirely but you could at least confirm the user entered a valid link (e.g. it ends in .gif)